### DISCUSSION TITLE
Enums as pivot column in relation manager

### DISCUSSION THREAD

**Discussion by neppoz** [filamentphp/filament#13703] [Help]
### Package

Other

### Package Version

v3.

### How can we help you?

Hi all. I try to use the enum column according to the documentation. I noticed that when used in a relation manager in the table, the badge does not set the color. On the other hand when editing the relation record, the Button with the right color is shown as expected.

Here some additional infos:

![image](https://github.com/user-attachments/assets/5887473f-50c5-433a-8032-cb2534470bac)
![image](https://github.com/user-attachments/assets/d3085948-1227-4b14-8c99-953fb52d4205)

The data is an BelongsToMany pivot table and in the models the enum is casted correctly.

**Comment by awcodes** [comment]
TextColumn::make('field')->badge()

If the field is cast as an enum it will just work.

BadgeColumn was deprecated in v3.

**Reply by neppoz** [reply]
Hi, thank you for your followUp. I used this as you suggested, the color of the badge stick to the primary color even if in the declaration of the enum the color is specified.

`Tables\Columns\TextColumn::make('rating_status')->badge(),`

```
    public function getColor(): string|array|null
    {
        return match ($this) {
            self::student => 'info',
            self::review => 'warning',
            self::rated => 'success',
        };
    }
```

Using the same concept on a Resource table works perfectly by the way.

**Reply by awcodes** [reply]
hmm.. looks ok to me.

**Reply by awcodes** [reply]
Can you shared the whole enum?

**Reply by neppoz** [reply]
yeah. Sure..

```
<?php

namespace App\Enums;

use Filament\Support\Contracts\HasColor;
use Filament\Support\Contracts\HasIcon;
use Filament\Support\Contracts\HasLabel;

enum RatingStatus: string implements HasColor, HasIcon, HasLabel
{
    case student = 'student';

    case review = 'review';

    case rated = 'rated';

    public function getLabel(): string
    {
        return match ($this) {
            self::student => 'Student',
            self::review => 'in Review',
            self::rated => 'Rated',
        };
    }

    public function getColor(): string|array|null
    {
        return match ($this) {
            self::student => 'info',
            self::review => 'warning',
            self::rated => 'success',
        };
    }

    public function getIcon(): ?string
    {
        return match ($this) {
            self::student => 'heroicon-m-sparkles',
            self::review => 'heroicon-m-arrow-path',
            self::rated => 'heroicon-m-check',
        };
    }
}
```

Here the snippet of the RelationManager declaration and the form and the table with the TextColumn

```
<?php

namespace App\Filament\Resources\UserResource\RelationManagers;

use App\Enums\RatingStatus;

class PlanesRelationManager extends RelationManager
...
...
...

public function form(Form $form): Form
    {
        return $form
            ->schema([
                Forms\Components\TextInput::make('callsign')
                    ->label('Callsign')
                    ->disabled(),
                Forms\Components\TextInput::make('base_price_per_minute')
                    ->label('Base price')
                    ->numeric()
                    ->required(),
                Forms\Components\TextInput::make('instructor_price_per_minute')
                    ->label('Instructor price')
                    ->numeric()
                    ->required(),
                Forms\Components\ToggleButtons::make('rating_status')
                    ->inline()
                    ->options(RatingStatus::class)
                    ->required(),
            ]);
    }

 public function table(Table $table): Table
    {
        return $table
            ->recordTitleAttribute('callsign')
            ->columns([
                Tables\Columns\TextColumn::make('callsign'),
                Tables\Columns\TextColumn::make('base_price_per_minute'),
                Tables\Columns\TextColumn::make('instructor_price_per_minute'),
                Tables\Columns\TextColumn::make('rating_status')->badge(),
            ])
...
...
```

**Reply by awcodes** [reply]
And you're positive that you have this on the model?

```php
protected $casts = [
   'rating_status' => RatingStatus::class
];
```

Everything looks ok to me. And it doesn't make sense that it wouldn't work in the RM if it's working in the normal resource. RM's are just using the Table package under the hood.

**Reply by miczed** [reply]
I literally had the same issue today. My problem was that my intermediate record was inheriting from `Model` instead of `Pivot`. This caused the casting of the enum to fail in the table. Make sure that the intermediate table that is casting the `RatingStatus` is extending the `Pivot` base class. Then everything works fine in the RelationManager.

**Reply by neppoz** [reply]
HI @miczed , can you share a code snippet for better understanding? Thanks
